extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/validation/ContractRequestValidator.java

```diff
@@ -40,4 +43,25 @@ public static Validator<JsonObject> instance() {
                 )
                 .build();
     }
+
+    /**
+     * This custom validator can be removed once `connectorAddress` is deleted and exists only for legacy reasons
+     */
+    private record MandatoryCounterPartyAddressOrConnectorAddress(JsonLdPath path) implements Validator<JsonObject> {
+
+        @Override
+        public ValidationResult validate(JsonObject input) {
+            var counterPartyAddress = new MandatoryValue(path.append(CONTRACT_REQUEST_COUNTER_PARTY_ADDRESS));
+            var validateCounterParty = counterPartyAddress.validate(input);
+            if (validateCounterParty.succeeded()) {
+                return ValidationResult.success();
+            }
+            var connectorAddress = new MandatoryValue(path.append(CONNECTOR_ADDRESS));
+            var validateConnectorAddress = connectorAddress.validate(input);
+            if (validateConnectorAddress.succeeded()) {
+                return ValidationResult.success();
```

Comment:
Please log a warning message that suggest to switch to the new attribute

Reply:
Added the warning message.